pkg/controller/certificates/signer/cfssl_signer.go

```diff
@@ -115,11 +119,21 @@ func (s *cfsslSigner) sign(csr *capi.CertificateSigningRequest) (*capi.Certifica
 	for _, usage := range csr.Spec.Usages {
 		usages = append(usages, string(usage))
 	}
+
+	certExpiryDuration := s.certificateDuration
+	durationUntilExpiry := s.ca.NotAfter.Sub(s.nowFn())
+	if durationUntilExpiry <= 0 {
+		return nil, fmt.Errorf("the signer has expired: %v", s.ca.NotAfter)
```

Comment:
this is technically a change (you can't get certs issued by the signing controller using an expired CA any more), but I see that as a good thing, preventing propagation of invalid certs that will just need to be replaced